tests/framework/extensions/rbac/verify.go

```diff
@@ -0,0 +1,252 @@
+func VerifyGlobalRoleBindings(t *testing.T, user *management.User, adminClient *rancher.Client){
+	query := url.Values{"filter": {"userName=" + user.ID}}
+	grbs, err := adminClient.Steve.SteveType("management.cattle.io.globalrolebinding").List(query)
+	require.NoError(t, err)
+	assert.Equal(t, 1, len(grbs.Data))
+}
+
+
+// VerifyListCluster validates a user with the required global permissions are able to/not able to list the clusters in rancher server
+func VerifyListCluster(t *testing.T, client, standardClient *rancher.Client, clusterID, role string) {
+	clusterList, err := standardClient.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+	require.NoError(t, err)
+	clusterStatus := &apiV1.ClusterStatus{}
+	err = v1.ConvertToK8sType(clusterList.Data[0].Status, clusterStatus)
+	require.NoError(t, err)
+
+	if role == restrictedAdmin {
+		adminClusterList, _ := client.Steve.SteveType(clusters.ProvisioningSteveResourceType).ListAll(nil)
+		assert.Equal(t, (len(adminClusterList.Data) - 1), len(clusterList.Data))
+	}
+	assert.Equal(t, 1, len(clusterList.Data))
+	actualClusterID := clusterStatus.ClusterName
+	assert.Equal(t, clusterID, actualClusterID)
+
+}
+// VerifyListProjects validates a user with the required cluster permissions are able/not able to list projects in the downstream cluster
+func VerifyListProjects(t *testing.T, client, standardClient *rancher.Client, clusterID, role, adminProjectName string) {
+	membersListProject, err := projects.ListProjectNames(standardClient, clusterID)
+	require.NoError(t, err)
+	projectlistAdmin, err := projects.ListProjectNames(client, clusterID)
+	require.NoError(t, err)
+
+	switch role {
+	case roleOwner, restrictedAdmin:
+		assert.Equal(t, len(projectlistAdmin), len(membersListProject))
+		assert.Equal(t, projectlistAdmin, membersListProject)
+	case roleMember:
+		assert.Equal(t, 0, len(membersListProject))
+	case roleProjectOwner, roleProjectMember:
+		assert.Equal(t, 1, len(membersListProject))
+		assert.Equal(t, adminProjectName, membersListProject[0])
+	}
+}
+
+// VerifyCreateProjects validates a user with the required cluster permissions are able/not able to create projects in the downstream cluster
+func VerifyCreateProjects(t *testing.T, client, standardClient *rancher.Client, clusterID, role string) {
+	memberProject, err := standardClient.Management.Project.Create(projects.NewProjectConfig(clusterID))
+	switch role {
+	case roleOwner, roleMember, restrictedAdmin:
+		require.NoError(t, err)
+		log.Info("Created project as a ", role, " is ", memberProject.Name)
+		require.NoError(t, err)
+		actualStatus := fmt.Sprintf("%v", memberProject.State)
+		assert.Equal(t, "active", actualStatus)
+	case roleProjectOwner, roleProjectMember:
+		require.Error(t, err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(t, "403 Forbidden", errorMsg[1])
```

Comment:
Would a const make sense for "403 Forbidden"?

Reply:
agreed

Reply:
yes it does as there are quite a few references to the forbidden Error. Updated